### DISCUSSION TITLE
Notify actions not overwrite defined default ones

### DISCUSSION THREAD

**Discussion by petr-panek** [quasarframework/quasar#14451] [Ideas / Proposals]
### What happened?

When default actions are defined with setDefaults, these action are always included in notify even when other action are defined.

### What did you expect to happen?

Action defined in $q.notify should overwrite default actions.

### Reproduction URL

https://codepen.io/petr-panek/pen/MWGmrMy

### How to reproduce?

1. Go to provided reprodution url

### Flavour

Quasar CLI with Vite (@quasar/cli | @quasar/app-vite)

### Areas

Plugins (quasar)

### Quasar info output

```Shell
Operating System - Darwin(21.6.0) - darwin/arm64
NodeJs - 16.16.0

Global packages
  NPM - 8.5.5
  yarn - 1.22.18
  @quasar/cli - 1.3.2
  @quasar/icongenie - 2.3.3
  cordova - Not installed

Important local packages
  quasar - 2.7.7 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app-vite - 1.0.6 -- Quasar Framework App CLI with Vite
  @quasar/extras - 1.15.3 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.38 -- The progressive JavaScript framework for building modern web UI.
  vue-router - 4.1.5
  pinia - Not installed
  vuex - 4.0.2 -- state management for Vue.js
  vite - 2.9.15 -- Native-ESM powered web dev build tool
  eslint - 8.22.0 -- An AST-based pattern checker for JavaScript.
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  *None installed*
```

**Comment by pbkompasz** [comment]
> Merge options into the default ones

**Comment by metalsadman** [comment]
Set `ignoreDefaults` prop of the config to true. ie.
`$q.notify({ ..., ignoreDefaults: true })`
https://github.com/quasarframework/quasar/blob/dev/ui/src/plugins/Notify.js#L146

**Reply by MilosPaunovic** [reply]
@metalsadman Do you think it would be a good idea to add an  `ignoreDefaults`  reference into the docs; I can't seem to find it anywhere at the moment.

**Reply by yusufkandemir** [reply]
@MilosPaunovic I don't know if you mean to add some sentences or code samples, but it's documented under the `create` method.
![image](https://user-images.githubusercontent.com/6266078/191446390-23ffd324-177a-4209-80f6-5930f3d2c492.png)

I think it would be nice if we add some mentions or code samples about it in https://quasar.dev/quasar-plugins/notify#setting-defaults though.

**Reply by petr-panek** [reply]
Shouldn't be solution based on "dependency injection" instead of "ignoreDefaults" prop?
It means if some property is specified in instance, it will overwrite the default one?
Thats's how Quasar normally works, right?

**Reply by Raymondo97** [reply]
The problem I have with `ignoreDefaults`, is that it ignores all defaults. There is no ability to choose which defaults to ignore.

**Reply by petr-panek** [reply]
> The problem I have with `ignoreDefaults`, is that it ignores all defaults. There is no ability to choose which defaults to ignore.

That's why the solution should be based on "config arrays merge" principle, injected properties will overwrite the default ones.
It should not be complicated to implement it with array destruction.

**Comment by Raymondo97** [comment]
Would there be any way to compare the array of objects in the actions? Maybe using an `Object.assign` or Lodash `lmergewith`? That way, if any actions objects have identical key value pairs, it doesn't repeat the button. Here are some examples of what I mean: 


<img width="553" alt="Screen Shot 2022-09-27 at 15 45 06" src="https://user-images.githubusercontent.com/91806440/192845583-07f0848a-87f1-4bb3-bc88-4b580f647f16.png">
<img width="282" alt="Screen Shot 2022-09-27 at 15 46 38" src="https://user-images.githubusercontent.com/91806440/192845585-d98f7349-ac73-46a0-93e8-571ab1478878.png">
<img width="304" alt="Screen Shot 2022-09-27 at 15 44 10" src="https://user-images.githubusercontent.com/91806440/192845576-06227f1f-5f47-4130-bea1-fe5e321bba2f.png">
<kbd><img width="282" alt="Screen Shot 2022-09-27 at 15 52 23" src="https://user-images.githubusercontent.com/91806440/192845586-6d6d1973-9c2d-43d4-a924-e71fcf88cbe6.png"></kbd>

## This is the resulting behavior that would be preferred.

<kbd><img width="282" alt="Screen Shot 2022-09-27 at 15 53 04" src="https://user-images.githubusercontent.com/91806440/192845589-e470e1a9-2e9a-433f-a96d-c6193e909b3d.png"></kbd>
